next.config.js

```diff
@@ -1,7 +1,23 @@
-const withPWA = require('next-pwa');
+const withPWA = require("next-pwa");
 
 module.exports = withPWA({
-    pwa: {
-        dest: 'public'
-    }
-});
\ No newline at end of file
+  trailingSlash: true,
+  async headers() {
+    return [
+      {
+        source: "/song/:slug*",
+        headers: [
+          {
+            key: "Cache-Control",
+            value: "public, max-age=604800, immutable",
```

Comment:
The **immutable** keyword will prevent us from changing the songs in the future, as the browsers will cache it for one year without possibility to flush it.

Reply:
Good point, @Vilican, should we just delete the immutable value? Will it not revalidate the cached files every time?

Reply:
@filiptronicek - It might be the solution. If there's no internet, then it will serve stale resource (will it?). Still, we can't lock ourselves from editing in the future by making it immutable.

Reply:
Agreed, I'll look more into the docs